docs/explanations/performance.md

```diff
@@ -0,0 +1,124 @@
+# Napari Performance
+
+Performance is a core feature of napari. Performance is a concern for every
+feature that napari implements. Without adequate performance users will migrate
+to other visualization tools, even if napari has impressive functionality.
+
+There are two main types of performance:
+
+**Objective Performance**
+* How long operations take when timed with a stopwatch.
+* Most times will vary based on the data being viewed.
+
+**Subjective Performance**
+* The user’s experience as it relates to performance.
+* Is the user’s experience pleasant or frustrating? Does napari "seem fast"?
+
+Both types of performance are important. No amount of slickness can make up for
+an application that is fundamentally too slow. And even a relatively fast
+application can feel clunky or frustrating if not designed well.
+
+## Objective Performance
+
+How to keep napari objectively fast:
+
+**Focus On Real Use Cases**
+
+* Focus on cases that matter to lots of people.
+* It’s easy to waste time optimizing things no one cares about or no one will
+  notice.
+* If a dataset is unreasonable or out of scope or fringe, don’t waste time
+  trying to make it run fast.
+
+**Always Be Timing**
+* Build timers into the software that always run.
+* If not always visible, power users and developers should be able to toggle them on.
+* This gives people an ambient awareness of how long things take.
+* Allows users to report concrete performance numbers:
+  *  *it seemed slow* &#8594; *it ran at 10Hz*.
+  *  *it took a long time* &#8594; *it took 2 minutes and 30 seconds*.
+* Teaches users how different hardware impacts performance.
```

Comment:
I was surprised to find out the other day that random seek times are two orders of magnitude faster on SSDs vs HDDs. Obviously I knew it was faster but I didn't know it was 100x faster!

The context was testing performance on a dataset on a laptop's SSD, where 121 1MB chunks worked almost equally fast as 1 121MB chunk. But when we applied it to the full data on an external HDD, the performance was unusable!

Reply:
It might be worth noting this fact here as an example.

Reply:
I added a line about HDD vs. SSD vs. network file systems.

Yes SSD seek times are so radically different there are algorithms today people run using SSD that aren't sped up versions of HDD algorithms, they are seek-intensive things that never would have been attempted on HDD. So it has opened some doors for having huge datastructures on disk that "hop around in" all over the place.

On the flip side if you do run one of those on HDD will performance so horrendously bad it will seem just broken and unusable.

It's also interesting the most common block storage for AWS system EBS is SSD based, but it's networked! You'd think if it's networked what does it matter if it's SSD or not, but you want/need the fast seek times even over the network.